.github/workflows/promote.yml

```diff
@@ -5,31 +5,95 @@ on:
+        description: Open a change case for this promotion. Only really needed for promotions to latest.
 
 jobs:
+  validate-inputs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Ensure either 'version' or 'old-channel' is passed"
+        if: ${{ !inputs.version && !inputs.old-channel }}
+        run: |
+          echo "ERROR: You must pass either 'version' or 'old-channel' to promote from."
+          echo "-> Use 'version' if you are promoting a new semver version to a channel. Example: I want to promote version '7.123.1' to channel 'dev'.
+          echo "-> Use 'old-channel' if you are promoting an existing channel to another channel. Example: I want to promote channel 'latest-rc' to channel 'latest'.
+          exit 1
+      - name: Ensure only one "starting point"
+        if: ${{ inputs.version && inputs.old-channel }}
+        run: |
+          echo "ERROR: Inputs 'version' and 'old-channel' cannot both be passed (exactlyOne)."
+          echo "-> Use 'version' if you are promoting a new semver version to a channel. Example: I want to promote version '7.123.1' to channel 'dev'.
+          echo "-> Use 'old-channel' if you are promoting an existing channel to another channel. Example: I want to promote channel 'latest-rc' to channel 'latest'.
+          exit 1
+      - name: Ensure channels do not include "stable"
+        if: ${{ contains(inputs.old-channel, 'stable') || ${{ contains(inputs.new-channel, 'stable') }}
+        run: |
+          echo "ERROR: Do not use 'stable' in channel names. Use 'latest' instead.
+          echo "-> S3 resources (stable, stable-rc) will be translated for you and will eventually be deprecated in favor of 'latest' and 'latest-rc'
+          exit 1
+
+  # Even if inputs.version is passed, we still need the SHA for a later step
+  get-package-info:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version-info.outputs.version }}
+      sha: ${{ steps.version-info.outputs.sha }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+          cache: yarn
+      - uses: salesforcecli/github-workflows/.github/actions/versionInfo@main
+        id: version-info
+        with:
+          version: ${{ inputs.version || inputs.old-channel }}
+
+  # Replace 'latest' with 'stable' for S3 resources
+  build-s3-channel:
+    runs-on: ubuntu-latest
+    needs: [get-package-info]
+    outputs:
+      s3-new-channel: ${{ steps.version-info.outputs.version }}
+    steps:
+      - run: |
+          S3_NEW_CHANNEL=${${{ inputs.new-channel }}/latest/stable}
+          echo "s3-new-channel=$S3_NEW_CHANNEL" >> $GITHUB_OUTPUT
+
   ctc-open:
-    if: inputs.useCTC
+    if: inputs.use-ctc
     uses: salesforcecli/github-workflows/.github/workflows/ctcOpen.yml@main
     secrets: inherit
 
   docker-promote:
-    needs: [ctc-open]
+    needs: [ctc-open, get-package-info]
```

Comment:
❓ If `ctc-open` is skipped because of its `if`, will this `needs` still be satisfied?

Reply:
@mshanemc this must work fine since this `needs` was not modified in this PR. The docker-promote was working prior

Reply:
No.  We've always created docker as latest-rc and a version simultaneously.  So I don't think the other path (no CTC) has ever been tested.